core/variant/typed_array.h

```diff
@@ -59,6 +59,20 @@ class TypedArray : public Array {
 	_FORCE_INLINE_ TypedArray() {
 		set_typed(Variant::OBJECT, T::get_class_static(), Variant());
 	}
+	_FORCE_INLINE_ TypedArray(const Vector<std::add_pointer_t<T>> &p_vector) {
+		resize(p_vector.size());
+		int offset = 0;
+		for (const std::add_pointer_t<T> &element : p_vector) {
```

Comment:
What's the `add_pointer_t` for? `T *` should work as well.

Reply:
For the constructor, that's technically true, but this was necessary for the iterator. I'm not *completely* sure why, but `const T *&element` fails to compile.

Reply:
This should compile: 
```c++
TypedArray(const Vector<T *> &p_vector) {
    resize(p_vector.size());
    int offset = 0;
    for (const T *&element : p_vector) {
        insert(offset++, element);
    }
}
```

Reply:
Setup some tests for these new constructors, and attempting to change to `T *` results in this error:
```
D:\Godot\Github\godot\core/variant/typed_array.h(74): error C2440: 'initializing': cannot convert from 'const T' to 'const T *&'
        with
        [
            T=Object *
        ]
        and
        [
            T=Object
        ]
D:\Godot\Github\godot\core/variant/typed_array.h(74): note: Conversion loses qualifiers
D:\Godot\Github\godot\core/variant/typed_array.h(74): note: the template instantiation context (the oldest one first) is
D:\Godot\Github\godot\tests/core/variant/test_array.h(665): note: see reference to class template instantiation 'TypedArray<Object>' being compiled
D:\Godot\Github\godot\core/variant/typed_array.h(71): note: while compiling class template member function 'TypedArray<Object>::TypedArray(const List<T *,DefaultAllocator> &)'
        with
        [
            T=Object
        ]
D:\Godot\Github\godot\tests/core/variant/test_array.h(709): note: see the first reference to 'TypedArray<Object>::TypedArray' in 'TestArray::DOCTEST_ANON_FUNC_5928'
```
I *think* it's interpreting `T`'s pointer as a type for the element itself, rather than keeping them isolated. `std::add_pointer_t<T>` seems to avoid this, so I'm sticking with it.